bin/calculate_TMB.py

```diff
@@ -0,0 +1,225 @@
+
+def popfrequency_filter(input, threshold):
+    TMB_popfreqfilt = input[~(input["CSQ_gnomADe_AF"] >= threshold)].reset_index(drop=True)
+    filtering_rates = len(TMB_popfreqfilt['Mut_ID'])
+    return(TMB_popfreqfilt, filtering_rates)
+
+def calculate_TMB(input, panel_size):
+    TMB = round((len(input.index)/panel_size)* 1000000, 2)
+    return(TMB)
+
+def plot_TMB(input, name_convention, lower_af, higher_af):
+    ### Preprocess for plotting
+    counts_consequence = input.groupby(input.columns.tolist(),as_index=False, dropna=False).size()
+    index_filter = (counts_consequence.groupby("Mut_ID")["size"].transform('max').ne(counts_consequence["size"]))
+    most_prevalent_con = counts_consequence[~index_filter.values]
+    TMB_forplot = pd.DataFrame({'Mut_ID': most_prevalent_con["Mut_ID"], 'CSQ_Consequence': most_prevalent_con["CSQ_Consequence"]}).merge(input).drop_duplicates()
+    TMB_forplot['CSQ_Consequence'] = TMB_forplot['CSQ_Consequence'].str.split('&').str[0] ## beautification
+    TMB_forplot = TMB_forplot.drop_duplicates("Mut_ID", keep='first') ### could be problematic, to discuss
+
+    ### Draw the plot
+    fig, ax = plt.subplots(figsize=(14, 8))
+
+    p = sns.histplot(data=TMB_forplot, ax=ax, stat="count", multiple="stack",
+                 binwidth=0.01,
+                 x="FORMAT_AF", kde=False, palette="colorblind",
+                 hue="CSQ_Consequence", element="bars")
+
+    ### Control the legend
+    sns.move_legend(p, loc='center right', bbox_to_anchor=(1.35, 0.5), ncol=1, fontsize=6)
+
+    ### Show filtering parameters
+    plt.axvline(x = lower_af, color = 'grey', linestyle = 'dotted')
+    plt.axvline(x = higher_af, color = 'grey', linestyle = 'dotted')
+
+    # Set the title and labels
+    ax.set_title('Allele frequency (AF) distribution for TMB-filtered variants')
+    ax.set_xlabel('AF in %')
+    ax.set_ylabel('# of mutations')
+    plt.xticks(np.arange(0, 1.1, 0.1))
+    ax.yaxis.get_major_locator().set_params(integer=True) ## force integers on y-axis
+    plt.savefig(name_convention, bbox_inches='tight')
+
+def main(argv=None):
+    """Coordinate argument parsing and program execution."""
+    args = parse_args(argv)
+    if not args.file_in.is_file():
+        logger.error(f"The given input file {args.file_in} was not found!")
+    TMB_df, filtering_rates_total = preprocess_vembraneout(args.file_in, args.filter_muttype.lower())
+    eligibility, panel_size = check_bed_size(args.bedfile, args.panelsize_threshold)
+    if eligibility == True:
+        TMB_covfilt, filtering_rates_cov = coverage_filter(TMB_df, args.min_cov)
+        filtering_rates_total.append(filtering_rates_cov)
+        TMB_affilt, filtering_rates_af = allelefrequency_filter(TMB_covfilt, args.min_AF, args.max_AF)
+        filtering_rates_total.append(filtering_rates_af)
+        TMB_popfilt, filtering_rates_popfreq = popfrequency_filter(TMB_affilt, args.popfreq_max)
+        filtering_rates_total.append(filtering_rates_popfreq)
```

Comment:
the single appends dont provide much value here,
i'd join it as lists `filtering_rates_total + [filtering_rates_cov, filtering_rates_af, ...]`